src/schema/fields/pagination.js

```diff
@@ -0,0 +1,116 @@
+import {
+  GraphQLBoolean,
+  GraphQLInt,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLNonNull,
+  GraphQLList,
+} from "graphql"
+import { toGlobalId } from "graphql-relay"
+
+const PREFIX = "arrayconnection"
+
+const PageCursor = new GraphQLObjectType({
+  name: "PageCursor",
+  fields: () => ({
+    cursor: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    page: {
+      type: new GraphQLNonNull(GraphQLInt),
+    },
+    isCurrent: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+    },
+  }),
+})
+
+export const PageCursorsType = new GraphQLObjectType({
+  name: "PageCursors",
+  fields: () => ({
+    first: {
+      type: PageCursor,
+      description:
+        "Optional, may be included in `around` (if current page is near the beginning).",
+    },
+    last: {
+      type: PageCursor,
+      description:
+        "Optional, may be included in `around` (if current page is near the end).",
+    },
+    around: {
+      type: new GraphQLNonNull(new GraphQLList(PageCursor)),
```

Comment:
The most correct way would be

```js
new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(PageCursor)))
```

(We should maybe start consider switching to SDL to define our schema.)